lib/server.js

```diff
@@ -204,11 +215,26 @@ class Server {
     };
 
     if (existingConnection) {
-      getSessionCallback(existingConnection.session);
-    } else if (sessionStorageProvider.isAsync()) {
-      sessionStorageProvider.get(sessionId, getSessionCallback);
+      const replacementTransport = connection.transport;
+      Object.keys(connection._transportListeners).forEach((event) => {
+        connection.transport.removeListener(
+          event, connection._transportListeners[event]
+        );
+      });
+      connection.transport = null;
+      connection.handshakeDone = true;
+      connection = existingConnection;
+      const cb = connection._onSessionRestored.bind(connection);
+      getSessionCallback(connection.session, replacementTransport, cb);
+      return;
+    }
+
+    if (sessionStorageProvider.isAsync()) {
+      sessionStorageProvider.get(sessionId, null, session =>
+        getSessionCallback(session, callback)
```

Comment:
It seems you have missed the second argument here (`session, null, callback`)?

Reply:
Yeah, thanks.